lib/pbench/server/database/models/datasets.py

```diff
@@ -1145,7 +1149,12 @@ def setvalue(dataset: Dataset, key: str, value: Any, user: Optional[User] = None
             except date_parser.ParserError as p:
                 raise MetadataBadValue(dataset, key, value, "date/time") from p
 
-            maximum = dataset.uploaded + __class__.config.max_retention_period
+            max_retention = ServerConfig.get(
+                key=ServerConfigOptions.MAXIMUM_DATASET_LIFETIME.key
+            )
+            maximum = dataset.uploaded + datetime.timedelta(
+                days=int(max_retention.value)
+            )
```

Comment:
I would be tempted to move some or all of the manipulations of the retention value to the assignment statement.  Black seems to molest this combination the least:
```
            max_retention = ServerConfig.get(
                key=ServerConfigOptions.MAXIMUM_DATASET_LIFETIME.key
            ).value
            maximum = dataset.uploaded + datetime.timedelta(days=int(max_retention))
```

Reply:
I don't actually find that any more palatable. If I were going for "less ugly" I'd probably break it up into at least three expressions. Maybe four.

Reply:
Ah, but then you would have to find names for all the intermediaries!  😉